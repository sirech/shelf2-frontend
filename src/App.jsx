import React, { Component } from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import { Container } from 'reactstrap'

import Header from './molecules/header'

class App extends Component {
  render () {
    return (
      <Router>
        <div>
          <Header />
          <Container fluid>
            <Route exact path='/' component={Home} />
          </Container>
        </div>
      </Router>
    )
  }
}

const Home = () => (
  <p className='App-intro'>
    To get started, edit <code>src/App.js</code> and save to reload.
  </p>
)

export default App
