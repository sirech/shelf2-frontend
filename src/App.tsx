import React, { Component } from 'react'
import {
  BrowserRouter as Router,
  Navigate,
  Route,
  Routes,
} from 'react-router-dom'

import Header from 'header'
import BooksArea from 'books_area'

import SearchList from 'search_list'
import NewBook from 'new_book'

import Callback from 'callback'

const year = new Date().getFullYear()

class App extends Component {
  render() {
    return (
      <Router>
        <div>
          <Header />
          <Content />
        </div>
      </Router>
    )
  }
}

const Content = () => (
  <Routes>
    <Route path="/" element={<Navigate replace to="/books" />} />
    <Route path="/books" element={<Navigate to={`/books/${year}`} />} />
    <Route path="/books/search" element={<Navigate replace to="/books" />} />
    <Route path="/books/search/:keyword" element={<SearchList />} />
    <Route path="/books/new" element={<NewBook />} />
    <Route path="/books/:year" element={<BooksArea />} />
    <Route path="/callback" element={<Callback />} />
  </Routes>
)

export default App
