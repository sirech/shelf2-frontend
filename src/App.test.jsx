import React from 'react'
import { fullRender } from 'test'

import App from './App'

jest.mock('rest/fetch')

describe('components', () => {
  describe('App', () => {
    it('renders correctly', () => {
      const { component } = fullRender(<App />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
