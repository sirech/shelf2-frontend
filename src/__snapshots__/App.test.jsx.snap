// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components App renders correctly 1`] = `
<BrowserRouter>
  <div>
    <Checks />
    <Header />
    <Content />
  </div>
</BrowserRouter>
`;
